scripts/wrist_test.py

```diff
@@ -0,0 +1,221 @@
+                # rospy.sleep(0.15)
+
+            # rospy.loginfo(msg)
+            # rospy.sleep(1.0)
+            # input()
+        else:
+            rospy.loginfo("down")
+            sh_axis1_targets = np.linspace(-1.0,-1.0, 5)
+            el_axis1_targets = np.linspace(ranges[-1], ranges[0], samples)
+
+            for t2 in el_axis1_targets:
+                msg.name = [ "elbow_"+lr+"_axis0"]
+                msg.position = [t2]
+                # print(t2)
+                msg.velocity = [0]
+                msg.effort = [0]
+                publisher.publish(msg)
+                rate.sleep()
+                # rospy.sleep(0.1)
+            # rospy.sleep(2.0)
+            # msg.position = [-0.3]
+
+
+def move_shoulder_elbow(lr="right", elbow=True):
+    msg = JointState()
+
+    msg.name = ["shoulder_"+lr+"_axis0"]
+    if elbow:
+        msg.name.append("elbow_"+lr+"_axis0")
+
+    targets = np.linspace(0.0, -1.1, 200)
+    targets_el = np.linspace(0, 0.6, 200)
+
+    if elbow:
+        for t, te in zip(targets,targets_el):
+            msg.position = [t, te]
+            msg.velocity = [0]*len(msg.name)
+            msg.effort = [0]*len(msg.name)
+            publisher.publish(msg)
+            rate.sleep()
+    else:
+        for t in targets:
+            msg.position = [t]
+        msg.velocity = [0]*len(msg.name)
+        msg.effort = [0]*len(msg.name)
+        publisher.publish(msg)
+        rate.sleep()
+
+
+def move_wrist(lr="right", axis=0):
+    msg = JointState()
+
+    ranges = [-1.5, -0.05]
+    samples = 200
+    msg.name = ["wrist_"+lr+"_axis"+str(axis), "wrist_"+lr+"_axis0"]
+    msg.velocity = [0]*len(msg.name)
+    msg.effort = [0]*len(msg.name)
+
+    out_str = lr + " " + str(axis) + " "
+
+    for i in range(6):
+
+        if not i % 2:
+            rospy.loginfo(out_str+"up")
+            targets = np.linspace(ranges[-1], ranges[0], samples)
+
+            for t2 in targets:
+                msg.position = [t2, t2]
+                publisher.publish(msg)
+                rospy.sleep(0.01)
+
+        else:
+            rospy.loginfo(out_str+"down")
+            targets = np.linspace(ranges[0], ranges[-1], samples)
+
+            for t2 in targets:
+                msg.position = [t2, t2]
+                publisher.publish(msg)
+                rospy.sleep(0.01)
+
+
+def move_all(lr="right", axis=0):
+    msg = JointState()
+
+    msg_name = ["shoulder_left_axis0", "elbow_left_axis0",  "wrist_left_axis0", "wrist_left_axis1",
+                "shoulder_right_axis0", "elbow_right_axis0",  "wrist_right_axis0", "wrist_right_axis1",
+                "head_axis0"
+                ]
+    # ["shoulder_"+lr+"_axis0", "elbow_"+lr+"_axis0",  "wrist_"+lr+"_axis0", "wrist_"+lr+"_axis1", "head_axis0"]
+    msg.name = msg_name
+    targets = np.linspace(0.0,-1.1, 200)
+    targets_el = np.linspace(0, 0.6, 200)
+    targets_wr = np.linspace(-1.5, -0.05, 200)
+    targets_head = np.linspace(-0.56, 0.56, 200)
```

Comment:
in the test all scene, we should move all joints in the range of the limits randomly

Reply:
Done